test/Sign.TestInfrastructure/ElevatedTheoryAttribute.cs

```diff
@@ -0,0 +1,26 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE.txt file in the project root for more information.
+
+using Xunit;
+
+namespace Sign.TestInfrastructure
+{
+    public sealed class ElevatedTheoryAttribute : TheoryAttribute
+    {
+        private const string SkipReason = "Test skipped because it requires elevation.";
+
+        private static readonly Lazy<bool> LazyIsElevated = new(IsElevated);
+
+        public override string Skip
```

Comment:
I'm curious about this approach. If for some reason the CI/CD doesn't run as elevated we'll miss the test coverage in an invisible way, right? I wonder if we should make this default to `SKIP = true` only when we're not running on CI/CD. I understand skipping automatically when running on devbox is nice. It would be confusing to contributors to run `dotnet test` on a git clone and immediately have failing tests. But we should never let these tests be skipped on CI/CD right?

One option I have used in the past is to detect our CI agent environment variable to do this "Is this CI" check. You could consider one of the variables here:
https://github.com/dotnet/reproducible-builds/blob/714df2e17493c71d0db054727f62a82810eeac7d/src/DotNet.ReproducibleBuilds/DotNet.ReproducibleBuilds.props#L15-L38

Reply:
Good feedback.  I updated it to execute if running in CI or elevated.